addon/utils/mung-options-for-fetch.ts

```diff
@@ -0,0 +1,49 @@
+import { assign } from '@ember/polyfills';
+import { serializeQueryParams } from './serialize-query-params';
+import {
+  Method,
+  FetchOptions,
+  AjaxOptions,
+  isPlainObject
+} from 'ember-fetch/types';
+
+/**
+ * Helper function that translates the options passed to `jQuery.ajax` into a format that `fetch` expects.
+ */
+export default function mungOptionsForFetch(
+  options: AjaxOptions
+): FetchOptions {
+  const hash = assign(
+    {
+      credentials: 'same-origin'
+    },
+    options
+  ) as FetchOptions;
```

Comment:
Do you remember why we end up doing this, instead of adding `credentials` to the `FetchOptions` type?

Reply:
So I need to return a `FetchOptions` type required by ts, `assign` doesn't automatically gives me this kind of type. Explicit declaration is needed. `credentials` is an optional field in `fetch` API, but we 100% should add this to `FetchOptions` type. Updated.